File: app/src/main/java/com/digital/playground/ui/dialog/DialogUtil.java
package com.digital.playground.ui.dialog;

import android.content.Context;
import android.content.DialogInterface;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AlertDialog;

import com.digital.playground.util.AppUtils;

public final class DialogUtil {
    private static AlertDialog mainDialog;

    /*
     * Show Loading Dialog
     * @param context the context
     */
    public static void show(Context context, String msg, @Nullable DialogInterface.OnClickListener onClickListener) {
        try {
            dismissDialog();
            mainDialog = new AlertDialog
                    .Builder(context)
                    .setMessage(msg)
                    .setTitle("Alert")
                    .setPositiveButton("OK", onClickListener)
                    .create();
            mainDialog.show();
        } catch (Exception e) {
            AppUtils.showException(e);
        }
    }


    /**
     * Dismiss existing Loading dialog.
     */
    public static void dismissDialog() {
        try {
            if (null != mainDialog && mainDialog.isShowing()) {
                mainDialog.dismiss();
            }
        } catch (Exception e) {
            AppUtils.showException(e);
        }
        mainDialog = null;

    }


}

File: app/src/main/java/com/digital/playground/ui/dialog/DialogUtil.kt
package com.digital.playground.ui.dialog

import android.content.Context
import android.content.DialogInterface
import androidx.appcompat.app.AlertDialog
import com.digital.playground.util.AppUtils.showException

/**
 * @Details DialogUtil
 * @Author Roshan Bhagat
 */
object DialogUtil {
    private var mainDialog: AlertDialog? = null

    /*
     * Show Loading Dialog
     * @param context the context
     */
    fun show(
        context: Context,
        msg: String,
        cancelable: Boolean = false,
        onClickListener: DialogInterface.OnClickListener? = null
    ) {
        try {
            dismissDialog()
            mainDialog = AlertDialog.Builder(context)
                .setMessage(msg)
                .setCancelable(cancelable)
                .setTitle("Alert")
                .setPositiveButton("OK", onClickListener)
                .create()
            mainDialog?.show()
        } catch (e: Exception) {
            showException(e)
        }
    }

    /**
     * Dismiss existing Loading dialog.
     */
    fun dismissDialog() {
        try {
            if (mainDialog?.isShowing == true) {
                mainDialog?.dismiss()
            }
        } catch (e: Exception) {
            showException(e)
        }
        mainDialog = null
    }
}